docs/Code-Conventions.md

```diff
@@ -139,6 +140,97 @@ The Checkstyle configuration file is located at `config/metrics/checkstyle/check
+
+In general, parameter substitution is a sufficient log guard.
+It also removes the need for explicit `toString()` calls.
+Parameter substitution can be applied to the above-mentioned non-guarded log statement as follows:
+```java
+log.debug("log something {} and {}", method(), param);
+```
+
+Unfortunately, PMD is currently subject to a [bug](https://github.com/pmd/pmd/issues/4703) that can lead to warnings still being reported despite the parametrized logging approach.
+If you utilized parameter substitution but are still facing the `GuardLogStatement` PMD warning, alternative approaches need to be found until the bug is resolved.
+These include local variable creation, suppression, and the fluent log API.
+For a decision about the appropriate approach, the context, the log level, and the individual performance impact of any relevant case should be taken into consideration.
+
+#### Local Variable Creation
+
+If the calculation performed was already performed before or will be performed after the log line (including in further log lines), introducing a local variable and referencing it in the log is a suitable approach.
+In some cases a respective local variable may already exist and simply be referenced in your log statement.
+```java
+String localVar = method();
+log.debug("log something {} and {}", localVar, param);
+[...]
+log.debug("log something else {}", localVar);
+```
+
+#### Suppression
+
+For Terasology complete disablement of the logging support is very unlikely, as is the disablement of the `error` log level.
+While `warn` and `info` log levels may be disabled in rare cases, they're usually not.
+Accordingly, any reported cases on `error`, `warn`, and `debug` log levels should be considered for suppression.
+
+Especially, if the performance impact is neglectable, e.g. for variable references (no method call) or simple getter or setter methods, or if the log frequency is very limited, e.g. only during initialization or cleanup, the PMD warning can be suppressed using an inline comment as follows:
+```java
+log.warn("log something {} and {}", method(), param); //NOPMD
+```
+
+If the logs are part of a logging-specific method, that is intentionally called (only) for logging specific aspects like machine specs or config values, the warning can be suppressed for the entire method like so:
+```java
+@SuppressWarnings("PMD.GuardLogStatement")
+public logSpecs() {
+log.info("log something {} and {}", method(), param)
+}
+```
+
+Suppressing warnings allows for easier identification and reversion once the PMD bug is resolved.
+
+#### Fluent Logging API
+
+If parameter substitution is insufficient, local variable creation is not suitable, and suppression is not desired, the fluent logging API can be utilized as follows:
+```java
+log.atDebug().log("log something {} and {}", method(), param);
+```
+
+Please do not use the more verbose variant(s) of the fluent logging API to keep complexity low and your log statements readable.
```

Comment:
I wanted to avoid this because I know people tend to skim through text and just copy anything :see_no_evil: 
That's why I didn't want to provide an example for the version we don't want.
If you think we need it though, I can add it. What do you think @BenjaminAmos ?

Reply:
yes, I think that's okay. I'll adjust it

Reply:
change it. is it alright like that?